src/model/entity/room.go

```diff
@@ -1,10 +1,82 @@
 package entity
 
+import (
+	"errors"
+	"math/rand"
+)
+
+type RoomType uint
+
+const (
+	RoomTypeStart RoomType = iota
+	RoomTypeOrdinary
+	RoomTypeFinish
+)
+
+const (
+	RoomMinHeight = 4
+	RoomMinWidth  = 4
+	RoomWalls     = 2
+)
+
+type TileType uint
+
 type Room struct {
 	Shape   Box
+	Type    RoomType
 	Foods   []Food
 	Elixirs []Elixir
 	Scrolls []Scroll
 	Weapons []Weapon
 	Enemies []Enemy
+	Portal  *Portal
+}
+
+func NewRoom(roomType RoomType, minX int, minY int, maxWidth int, maxHeight int) (*Room, error) {
```

Comment:
Вот тут можно упростить список аргументов до:

```suggestion
func NewRoom(type RoomType, position Point2D[int], maxSize Size2D[uint]) (*Room, error) {
```